webapi/src/main/java/org/sakaiproject/webapi/controllers/GradesController.java

```diff
@@ -66,103 +69,158 @@ public class GradesController extends AbstractSakaiApiController {
     private org.sakaiproject.grading.api.GradingService gradingService;
 
     @Resource
-    private SiteService siteService;
-
-    private GradingService samigoGradingService;
-
-    private Function<Site, List<GradeRestBean>> convert = (s) -> {
-
-        try {
-            ToolConfiguration tc = s.getToolForCommonId("sakai.gradebookng");
-            String gbUrl = tc != null ? "/portal/directtool/" + tc.getId() : "";
-            return gradingService.getAssignments(s.getId()).stream()
-                .map(a -> {
+    private PortalService portalService;
 
-                    try {
-                        GradeRestBean gtb = new GradeRestBean(a);
-
-                        gtb.setSiteTitle(s.getTitle());
-
-                        List<String> students = new ArrayList<>(s.getUsers());
-                        List<GradeDefinition> grades
-                            = gradingService.getGradesForStudentsForItem(s.getId(), a.getId(), students);
+    @Resource
+    private SiteService siteService;
 
-                        double total = 0;
-                        for (GradeDefinition gd : grades) {
-                            if (Objects.equals(GradingConstants.GRADE_TYPE_POINTS, gd.getGradeEntryType())) {
-                                String grade = gd.getGrade();
-                                if (!StringUtils.isBlank(grade)) {
-                                    total += Double.parseDouble(grade);
+    private final Function<String, List<GradeRestBean>> gradeDataSupplierForSite = (siteId) -> {
+        List<org.sakaiproject.grading.api.Assignment> assignments = gradingService.getViewableAssignmentsForCurrentUser(siteId);
+        List<Long> assignmentIds = assignments.stream().map(org.sakaiproject.grading.api.Assignment::getId).collect(Collectors.toList());
+
+        // no need to continue if the site doesn't have gradebook items
+        if (!assignmentIds.isEmpty()) {
```

Comment:
Why not return here and reduce the indentation by a tab? It's hard to see the column number in a diff though.